MiniShopping/MiniShopping/Controllers/ProductsViewController.m

```diff
@@ -0,0 +1,86 @@
+#import "ProductsViewController.h"
+#import "ProductsView.h"
+#import "ProductCell.h"
+#import "Product.h"
+#import "FetchProducts.h"
+#import "NSMutableArray+Sorting.h"
+
+
+@interface ProductsViewController () <UICollectionViewDelegate, UICollectionViewDataSource, FetchProductsDelegate, ProductViewDelegate> {
+    NSMutableArray<Product *> *products;
+    ProductsView *productsView;
+    FetchProducts *fetchProducts;
+}
+
+@end
+
+@implementation ProductsViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.title = @"Mini Shopping";
+    self.view.backgroundColor = UIColor.systemBackgroundColor;
+    [self setupModel];
+    [self setupView];
+}
+
+- (void)setupModel {
+    products = [[NSMutableArray alloc] init];
+    fetchProducts = [[FetchProducts alloc] init];
+    [fetchProducts fetchData];
+    fetchProducts.delegate = self;
+}
+
+- (void)setupView {
+    productsView = [[ProductsView alloc] initWithFrame:CGRectZero];
+    productsView.delegate = self;
+    [productsView.productsGridView registerClass:[ProductCell class] forCellWithReuseIdentifier:@"productsGridViewIdentifier"];
+    productsView.productsGridView.translatesAutoresizingMaskIntoConstraints = NO;
+    productsView.productsGridView.delegate = self;
+    productsView.productsGridView.dataSource = self;
+    [self.view addSubview:productsView];
+    
+    productsView.translatesAutoresizingMaskIntoConstraints = NO;
+    [productsView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor].active = YES;
+    [productsView.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor].active = YES;
+    [productsView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor].active = YES;
```

Comment:
Leading and trailing constraints should also be set w.r.t safe area layout. Because in landscape orientation, leading/trailing side view may be obscured by device notch/home indicator bar.